hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -752,6 +752,7 @@ public void dropMetadataPartitions(List<MetadataPartitionType> metadataPartition
       LOG.warn("Deleting pending indexing instant from the timeline for partition: " + partitionPath);
       deletePendingIndexingInstant(dataMetaClient, partitionPath);
     }
+    closeInternal();
   }
 
```

Comment:
Can we let the `HoodieBackedTableMetadataWriter` implement `AutoClosable` so that it can be used in the try finally block. Can we also fix `HoodieFlinkWriteClient#writeTableMetadata`,

Reply:
`HoodieBackedTableMetadataWriter` extends `HoodieTableMetadataWriter` which implements `AutoClosable`. But yeah we could create the writer in try-with-resource.

+1 for fixing `HoodieFlinkWriteClient` too.